haystack_experimental/components/retrievers/multi_query_embedding_retriever.py

```diff
@@ -0,0 +1,151 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from concurrent.futures import ThreadPoolExecutor
+from typing import Any, List, Optional
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.components.embedders.types.protocol import TextEmbedder
+from haystack.core.serialization import component_to_dict
+from haystack.utils.deserialization import deserialize_component_inplace
+
+from haystack_experimental.components.retrievers.types import EmbeddingRetriever
+
+
+@component
+class MultiQueryEmbeddingRetriever:
```

Comment:
We should add warm_up methods to both of these components, which then calls the warm_up methods of the components they accept in their init methods. For example, here is the one we use in `Agent`
```python
    def warm_up(self) -> None:
        """
        Warm up the Agent.
        """
        if not self._is_warmed_up:
            if hasattr(self.chat_generator, "warm_up"):
                self.chat_generator.warm_up()
            self._is_warmed_up = True
```

Reply:
right, I had something similar as just code in the `init()` - but this is the way yes